tfx/extensions/google_cloud_ai_platform/trainer/executor.py

```diff
@@ -97,6 +104,9 @@ def Do(self, input_dict: Dict[str, List[types.Artifact]],
       absl.logging.error(err_msg)
       raise ValueError(err_msg)
 
+    job_args['labels'] = { **job_args.get('labels', {}),
+                           **custom_config.get(LABELS_KEY, {}) }
```

Comment:
Let's pass it in just like job_id (line 110)? or do it like [project](https://github.com/tensorflow/tfx/blob/master/tfx/extensions/google_cloud_ai_platform/runner.py#L191)

as the [job_args](https://github.com/tensorflow/tfx/blob/f84b768f28ed60ab1b51dded73c64718950db1f4/tfx/extensions/google_cloud_ai_platform/training_clients.py#L200) of client.create_training_job is the training_input, so I think it would be good to keep it as it is, and add an optional labels to that function?

Reply:
1. If it is passed in like `job_id`, then the signatures of `runner.start_cloud_training`, `CAIPJobClient.create_training_job` and `VertexJobClient.create_training_job` would have to change, and it seems a bit convoluted, since `labels` is part of `job_args` for `VertexJobClient` already?
2. Similar argument for doing it like `project`, but it would be slightly simpler than option 1 since you would not have to change `runner.start_cloud_training` in addition.

I see your point about the `job_args` in `CAIPJobClient` should not have `labels`, but it is simply stored there and [popped](https://github.com/axeltidemann/tfx/blob/18f05ff652919f3e8566afd583e4fb860e29b392/tfx/extensions/google_cloud_ai_platform/training_clients.py#L230) before passed on - which is similar to the "stowaway" project you refer to. I personally think my proposal is the least "intrusive" option, but I have implemented option 1 for you to review.

Reply:
I see your point. if we want to just pass the label in job_args as a workaround for CAIP, then we don't need to add constant to executor.py, those constants are top level keys of custom_config

Reply:
So you mean like my first commit, basically?

Reply:
I will go on holiday leave now, and will check back in two weeks. If I understand you correctly, what you are describing is my first commit - feel free to approve that in the meantime. If not, I am of course happy to continue this when I am back. Happy holidays!

Reply:
Current code look good (except we want to make labels an optional args for AI platform only)

(Due to Vertex/AI platform diff, I don't think we have a perfect way to align them)

Reply:
I have made `job_labels` optional, please review.